test/testdata/lsp/code_actions/move_method/class_func.A.rbedited

```diff
@@ -0,0 +1,47 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module GreetingModule
+  extend T::Sig
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Move method to a new module
+    'Hello'
+  end
+end
+
+module Foo
+  extend T::Sig
+
+  
+  
+  
+  sig do
+    params(x: String)
+    .returns(String)
+  end
+  def name(x)
+    "#{GreetingModule.greeting} #{x}"
+  end
+end
+
+module A
+  class B
+    extend T::Sig
+
+    sig {void}
+    def bar
+      m = Foo
+      GreetingModule.greeting
+
+      GreetingModule.greeting
+      print((GreetingModule).greeting)
```

Comment:
Is it intentional that we dropped the `if true` that was present in the original?

Reply:
Yes, if the rename location is the send location we don't analyze the receiver, but replace the whole receiver with the new module

https://github.com/sorbet/sorbet/blob/f895091951e3ba59a162834299e282383acae231/main/lsp/requests/code_action.cc#L174-L176

Reply:
Ahh, ok I see what's going on. Sorbet is smart enough to see that the condition is always `true` so that the receiver is always the same type.

In that case I have another test case for you:

```ruby
# typed: true
extend T::Sig

class Left
  def self.foo; end
end

class Right
  def self.foo; end
end

sig {params(x: T.any(Left, Right)).void}
def example(x)
  x.foo
end
```

Let's say that the user requested to rename **only** the `def self.foo; end` in `Left`. I think that for cases like this, we should decline to rename the `x.foo` call site. Instead we should just let Sorbet retypecheck the codebase, which will cause it to report a type error, because it'll have to be something that the user manually fixes, not something that can be systematically renamed.

To do this, you should be able to look in the `SendResponse::dispatchResult`. If you look at the structure of the `DispatchResult` object, you'll see it has a `main` DispatchComponent` and a `secondary` DispatchResult, which basically makes `dispatchResult` be the start of a linked list of DispatchResults. This linked list will have one entry for each method that the call site dispatched to, which in the `x.foo` case will have one entry for `Left.foo` and one entry for `Right.foo`.

We can use that information in one of a number of somewhat equivalent ways to detect this case:

- Loop over the chain of DispatchResults, and assert for all of them that `->main.method` is the method that the move request was called on, or
- Don't bother with looping, just assert that `dispatchResult->main.method` is the method we're moving and that `dispatchResult->secondary == nullptr` (i.e., that the linked list is length 1).

lmk if that's confusing, happy to chat more.

Reply:
Wow, what a catch! I could even think of that use case, thanks for flagging! I'll limit call site updates only to trivial use cases and add a test